### ISSUE TITLE
PopScope in Navigator 2.0 doesn't call onPopInvoke when canPop is true

### ISSUE THREAD

**Issue by michalowskim** [flutter/flutter#143753]
### Steps to reproduce

I've ran into strange `PopScope` behaviour and I believe it should not work this way. It only happens while using Navigator 2.0 - with Navigator 1.0 `PopScope` works as expected. I've tested it on both Android and iOS and outputs were the same.

`PopScope` doesn't trigger `onPopInvoked` when `canPop` is set to `true`

Steps to reproduce:
1. Make a page with a `PopScope` widget with `canPop: true`
2. Navigate to this page with Navigator 2.0 (go_router or auto_route)
3. Pop the page

### Expected results

`onPopInvoked` callback in `PopScope` is called with `didPop = true` 

### Actual results

`onPopInvoked` is never called

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:go_router/go_router.dart';

void main() {
  runApp(const MyApp());
}

final GoRouter _router = GoRouter(
  routes: <RouteBase>[
    GoRoute(
      path: '/',
      builder: (BuildContext context, GoRouterState state) {
        return const MyHomePage();
      },
      routes: <RouteBase>[
        GoRoute(
          path: 'details',
          builder: (BuildContext context, GoRouterState state) {
            return const SecondPage();
          },
        ),
      ],
    ),
  ],
);

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp.router(
      title: 'Flutter Demo',
      routerConfig: _router,
    );
  }
}

class MyHomePage extends StatelessWidget {
  const MyHomePage({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text("First Page"),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            ElevatedButton(
              onPressed: () => context.go('/details'),
              child: const Text('Go to second page'),
            ),
          ],
        ),
      ),
    );
  }
}

class SecondPage extends StatelessWidget {
  const SecondPage({super.key});

  @override
  Widget build(BuildContext context) {
    return PopScope(
      canPop: true,
      onPopInvoked: (bool didPop) {
        print("didPop: $didPop");
      },
      child: Scaffold(
        appBar: AppBar(
          title: const Text("Second Page"),
        ),
      ),
    );
  }
}
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.16.9, on macOS 14.0 23A344 darwin-arm64, locale en-PL)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0-rc3)
[✓] Xcode - develop for iOS and macOS (Xcode 15.2)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2023.1)
[✓] VS Code (version 1.84.2)
[✓] Connected device (3 available)
[✓] Network resources

• No issues found!
```

</details>

**Comment by dipesh-panchal** [comment]
Was able to reproduce with above code on Flutter 3.16.9

**Comment by darshankawar** [comment]
@michalowskim 
This seems to be similar to https://github.com/flutter/flutter/issues/141189 which was fixed recently. The fix doesn't seem to be in latest stable 3.19, but can be verified on latest master. 

`flutter channel master`
`flutter upgrade`

I'll close this as it is fixed already. If you disagree, write in comments and I'll reopen it.
